lib/valueflow.cpp

```diff
@@ -1658,8 +1658,79 @@ static bool evalAssignment(ValueFlow::Value &lhsValue, const std::string &assign
     return true;
 }
 
+template<class T>
+struct SingleRange {
+    T* x;
+    T* begin() const {
+        return x;
+    }
+    T* end() const {
+        return x+1;
+    }
+};
+
+template<class T>
+SingleRange<T> MakeSingleRange(T& x)
+{
+    return {&x};
+}
+
+class MultiRange {
+    using M = std::unordered_map<nonneg int, ValueFlow::Value>;
+
+    struct Iterator {
+        using value_type = const ValueFlow::Value;
+        using pointer = value_type *;
+        using reference = value_type &;
+
+        explicit Iterator(const M::const_iterator &it)
+            : mIt(it) {
+        }
+
+        reference operator*() const {
+            return mIt->second;
+        }
+
+        pointer operator->() {
+            return &mIt->second;
+        }
+
+        Iterator &operator++() {
```

Comment:
Post increment is required for forward iterators.

Reply:
done

Reply:
This change leads to a Cppcheck warning:
```
lib/valueflow.cpp:1704:13: performance: Prefer prefix ++/-- operators for non-primitive types. [postfixOperator]
            mIt++;
            ^
```

Maybe this needs to be checked the other way around for this case.

Reply:
I suppressed this for now.